feat: integrate Actual API for transaction storage and management

baruchiro opened the pull request.

baruchiro committed:
feat: integrate Actual API for transaction storage and management

autofix-troubleshooter commented:
Hi! I'm the <a href="https://autofix.ci"><img alt="autofix logo" src="https://autofix.ci/logo/logo.svg" width="16"><strong>autofix.ci</strong></a> troubleshooter bot.

It looks like you correctly set up a CI job that uses the autofix.ci GitHub Action, but the [**autofix.ci GitHub App**](https://github.com/apps/autofix-ci/) has not been installed for this repository. This means that autofix.ci unfortunately does not have the permissions to fix this pull request. If you are the repository owner, please install the app and then restart the CI workflow! 😃

baruchiro committed:
update README

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

baruchiro committed:
add it to the list

baruchiro committed:
fix: improve directory existence check in ActualBudgetStorage

baruchiro committed:
fix: update date type in ActualTransaction and format date in ActualBudgetStorage

baruchiro committed:
feat: add account name mapping to ActualBudgetStorage for improved logging

baruchiro committed:
chore: update @actual-app/api to version 25.5.0 and bump dependencies

baruchiro committed:
Merge remote-tracking branch 'origin/main' into feat/actual-budget

# Conflicts:
#	package-lock.json
#	package.json
#	src/bot/storage/index.ts

baruchiro committed:
coderabbit suggestions

baruchiro committed:
coderabbit fixes

baruchiro committed:
coderabbit fix

baruchiro committed:
Handle errors during Actual Budget initialization and improve error messages

dependabot[bot] committed:
chore(deps): bump israeli-bank-scrapers from 5.4.7 to 6.0.0 (#461)

* chore(deps): bump israeli-bank-scrapers from 5.4.7 to 6.0.0

Bumps [israeli-bank-scrapers](https://github.com/eshaham/israeli-bank-scrapers) from 5.4.7 to 6.0.0.
- [Release notes](https://github.com/eshaham/israeli-bank-scrapers/releases)
- [Commits](https://github.com/eshaham/israeli-bank-scrapers/compare/v5.4.7...v6.0.0)

---
updated-dependencies:
- dependency-name: israeli-bank-scrapers
  dependency-version: 6.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Rename israeli-bank-scrapers+5.4.7.patch to israeli-bank-scrapers+6.0.0.patch

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Daniel Hauser <36034483+daniel-hauser@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump the development-dependencies group with 2 updates (#463)

Bumps the development-dependencies group with 2 updates: [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) and [ts-jest](https://github.com/kulshekhar/ts-jest).


Updates `@types/node` from 22.15.17 to 22.15.18
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

Updates `ts-jest` from 29.3.2 to 29.3.3
- [Release notes](https://github.com/kulshekhar/ts-jest/releases)
- [Changelog](https://github.com/kulshekhar/ts-jest/blob/main/CHANGELOG.md)
- [Commits](https://github.com/kulshekhar/ts-jest/compare/v29.3.2...v29.3.3)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-version: 22.15.18
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
- dependency-name: ts-jest
  dependency-version: 29.3.3
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

daniel-hauser committed:
upgrade israeli-bank-scrapers@6.1.2 (#465)


* israeli-bank-scrapers@6.1.2

* fix patch

dependabot[bot] committed:
chore(deps): bump debug from 4.4.0 to 4.4.1 (#462)

Bumps [debug](https://github.com/debug-js/debug) from 4.4.0 to 4.4.1.
- [Release notes](https://github.com/debug-js/debug/releases)
- [Commits](https://github.com/debug-js/debug/compare/4.4.0...4.4.1)

---
updated-dependencies:
- dependency-name: debug
  dependency-version: 4.4.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

baruchiro committed:
Merge remote-tracking branch 'upstream/main' into feat/actual-budget

# Conflicts:
#	package-lock.json

doriaviram committed:
feat(scraper): Make additionalTransactionInformation configurable via env var (#464)

* feat(scraper): Make additionalTransactionInformation configurable via ENV var

* [autofix.ci] apply automated fixes

* Adding `additionalTransactionInformationEnabled` to github scrape action

* `npm run lint:fix`

* fix: update type to boolean

* refactor: update docker run command formatting for readability

---------

Co-authored-by: autofix-ci[bot] <114827586+autofix-ci[bot]@users.noreply.github.com>
Co-authored-by: Daniel Hauser <36034483+daniel-hauser@users.noreply.github.com>
Co-authored-by: Daniel Hauser <dd.hauser@gmail.com>

dependabot[bot] committed:
chore(deps-dev): bump the development-dependencies group across 1 directory with 2 updates (#468)

Bumps the development-dependencies group with 2 updates in the / directory: [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) and [pretty-quick](https://github.com/prettier/pretty-quick).


Updates `@types/node` from 22.15.19 to 22.15.29
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

Updates `pretty-quick` from 4.1.1 to 4.2.0
- [Release notes](https://github.com/prettier/pretty-quick/releases)
- [Changelog](https://github.com/prettier/pretty-quick/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prettier/pretty-quick/compare/v4.1.1...v4.2.0)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-version: 22.15.29
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development-dependencies
- dependency-name: pretty-quick
  dependency-version: 4.2.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nisan-tagar committed:
chore(deps): bump buxfer client to 1.1.4 (#469)

* update scrapper versrion, and work in progress on buxfer integration

* Added Buxfer API integration

* Added Buxfer API integration

* Change `package-lock.json` to contain only the `buxfer-ts-client` change

* Pass the Buxfer environment variables to the scrape workflow

* Added Buxfer API integration - bump to 1.0.4

* Update src/storage/buxfer.ts

Co-authored-by: Daniel Hauser <36034483+daniel-hauser@users.noreply.github.com>

* Update scrape.yml

specify the environment of the scrape action to have better secret env management

* Update scrape.yml

* bump buxder client to 1.1.3 to better support deduplicated and ignored transactions that should not be uploaded due to API limitations ...

* bump buxfer client to 1.1.3

* bump "buxfer-ts-client": "^1.1.4",

---------

Co-authored-by: Daniel Hauser <36034483+daniel-hauser@users.noreply.github.com>
Co-authored-by: Daniel Hauser <dd.hauser@gmail.com>

dependabot[bot] committed:
chore(deps): bump @mswjs/interceptors from 0.38.6 to 0.38.7 (#467)

Bumps [@mswjs/interceptors](https://github.com/mswjs/interceptors) from 0.38.6 to 0.38.7.
- [Release notes](https://github.com/mswjs/interceptors/releases)
- [Changelog](https://github.com/mswjs/interceptors/blob/main/release.config.json)
- [Commits](https://github.com/mswjs/interceptors/compare/v0.38.6...v0.38.7)

---
updated-dependencies:
- dependency-name: "@mswjs/interceptors"
  dependency-version: 0.38.7
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Daniel Hauser <36034483+daniel-hauser@users.noreply.github.com>

daniel-hauser committed:
Merge branch 'main' into feat/actual-budget

daniel-hauser committed:
`git checkout main -- .\package-lock.json`

daniel-hauser committed:
`npm install`
